Warn when live update pass exceeds refresh_interval

skprasadu opened the pull request:
fixes #267

## What
Adds a warning when an interval live update pass takes longer than refresh_interval,
so interval updates will lag behind.

## Why
When refresh_interval is small, users may not realize the updater is falling behind.
This warning makes lag visible.

## How
- Added warn_if_overrun(fut, threshold, warn_fn) helper
- Wired threshold into interval update passes only (not initial pass)
- Keeps MissedTickBehavior::Delay

## Testing
- cargo test (unit tests in live_updater.rs)
- dev/run_cargo_test.sh warn_if_overrun
- E2E: examples/live_updates with a large file triggers WARN:
  "Live update pass exceeded refresh_interval; interval updates will lag behind"

## Notes
- `cargo clippy -p cocoindex --all-targets -- -D warnings` fails on main due to
  `clippy::needless_return` in `rust/utils/src/batching.rs:218` (unrelated).

georgeh0 commented on rust/cocoindex/src/execution/live_updater.rs:
```diff
@@ -424,13 +447,40 @@ impl SourceUpdateTask {
+                source_name = %source_name,
+                update_title = %update_title_owned,
+                refresh_interval_secs = threshold.as_secs_f64(),
+                elapsed_secs = elapsed.as_secs_f64(),
```
these two are expected to be the same. We only need to log one.

skprasadu commented:
@georgeh0 , updated to move the overrun warning into the interval tick loop (tokio::select! between update and sleep(refresh_interval)), removed the helper, and log only refresh_interval_secs (no elapsed_secs). Confirmed end-to-end by appending to a large file and verifying Postgres row length/tail updates; warning appears when refresh interval is too small. I test it.

georgeh0 reviewed (approved):
Awesome! Thanks for the PR!
